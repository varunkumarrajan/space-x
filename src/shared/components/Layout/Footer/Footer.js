// React
import React from 'react';
// Prop types
import PropTypes from 'prop-types'; // ES6
// Styled component
import styled from 'styled-components';

// Styled footer wrapper
const FooterWrapper = styled.footer`
  font-size: 16px;
  font-weight: bold;
  text-align: center;
`;

// Styled footer small wrapper
const FooterSmallWrapper = styled.div`
  display: inline-block;
`;

/**
 * @description html semantic Footer
 * @param {object} {content}
 */
const Footer = ({ content }) => (
  <FooterWrapper data-testid="footerId">
    {content.title}: <FooterSmallWrapper data-testid="footerChildId">{content.name}</FooterSmallWrapper>
  </FooterWrapper>
);

// Default footer
export default Footer;

// Default props
Footer.defaultProps = {
  content: {
    title: '',
    name: '',
  },
};

// Prop types
Footer.propTypes = {
  content: PropTypes.shape({
    title: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
};
