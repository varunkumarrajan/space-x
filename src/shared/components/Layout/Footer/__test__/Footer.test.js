// React
import React from 'react';
// Testing library
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import Footer from './../Footer';

// Mock data
const footerMock = {
    title: 'test',
    name: 'testing'
}

afterEach(cleanup);

describe('Footer', () => {
  it('Should Render Footer with props', () => {
    const { getByTestId } = render(<Footer content={footerMock} />);
    const footerId = getByTestId('footerId');
    const footerChildId = getByTestId('footerChildId');
    expect(footerId.children).not.toHaveLength(0);
    expect(footerId.className).toBeTruthy();
    expect(footerChildId.className).toBeTruthy();
  });

  it('Should footer Snapshot', () => {
    const footer = renderer.create(<Footer content={footerMock} />).toJSON();
    expect(footer).toMatchSnapshot();
  });
});
