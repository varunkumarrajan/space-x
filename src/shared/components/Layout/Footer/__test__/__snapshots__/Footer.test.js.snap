// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should footer Snapshot 1`] = `
<footer
  className="sc-bdfBwQ gdBJkk"
  data-testid="footerId"
>
  test
  : 
  <div
    className="sc-gsTCUz fhPuTa"
    data-testid="footerChildId"
  >
    testing
  </div>
</footer>
`;
